### PR TITLE
Add withAddedLink()

### PR DESCRIPTION
**Pull request by mosaxiv**
https://github.com/cakephp/cakephp/issues/11698

### CODE REVIEW

File: src/Http/Response.php [hunk 1]
```diff
@@ -1983,6 +1984,44 @@ public function withLength($bytes)
...
+     * Link: </posts/view>; rel="next"
+     * ```
+     *
+     * @param string|array $url link-value
+     * @param array $options link-param
+     * @return static
+     */
+    public function withAddedLink($url, $options = [])
+    {
+        $url = '<' . Router::url($url) . '>';
```
Comment (markstory) [inline src/Http/Response.php]: Would be preferable to only accept string URLs in this method. Right now the options to `Router::url()` cannot be used.

File: src/Http/Response.php [hunk 2]
```diff
@@ -1983,6 +1983,42 @@ public function withLength($bytes)
...
+     * ```
+     *
+     * Will generate:
+     *
+     * ```
+     * Link: <http://example.com?page=1>; rel="prev"
+     * Link: <http://example.com?page=3>; rel="next"
+     * ```
+     *
+     * @param string $url link-value
```
Comment (ADmad) [inline src/Http/Response.php]: Can we have a better descriptive statement instead of `link-value` please? :)  Same for options param.

Comment (ADmad) [reply src/Http/Response.php]: And `@since 3.6.0` tag.

Comment (mosaxiv) [reply src/Http/Response.php]: hmmm :anguished:
Sorry, I can't think of good ideas...
If revision is required, I want you to help me.

**Pull request merged**